### PR TITLE
CHI-1486: Migrate last few pieces of runtime JS to TS

### PR DESCRIPTION
**Pull request by stephenhand**
Primary Reviewer: @murilovmachado or @robert-bo-davis 

## Description

* Migrated last few utility functions, tests and test support modules still in JS to TS
* Removed the notion of 'per environment' configuration (it was not being used anyway)
* Flattened 'routes' module into single, top level file
* Fixed the `start:service` script

### Checklist
- [X] Corresponding issue has been opened
- [X] New tests added

### Related Issues
CHI-1486

### Verification steps

Should be verified by automated tests

### CODE REVIEW

File: hrm-service/src/config/db.ts [hunk 1]
```diff
@@ -1,17 +1,12 @@
...
+env.config();
+
+export default {
   username: process.env.RDS_USERNAME || 'hrm',
   password: process.env.RDS_PASSWORD || null,
   database: process.env.RDS_DBNAME || 'hrmdb',
   host: process.env.RDS_HOSTNAME || 'localhost',
   port: process.env.POSTGRES_PORT || 5432,
   dialect: 'postgres',
 };
```
Comment (GPaoloni) [inline hrm-service/src/config/db.ts:12]: Isn't this identical to `hrm-service/db-migrate.js`. Maybe we could rely on one of them only?

File: hrm-service/src/search.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
+const { OrderByDirection } = require('./sql');
```
Comment (GPaoloni) [inline hrm-service/src/search.ts:1]: Can't this be an `import`?

**Review by GPaoloni — APPROVED** [review]
Awesomeee, thanks!! :rocket: 
Haven't run the code, but since it's just moving stuff around, I'll trust the CI :P

File: hrm-service/src/config/db.ts (hunk 1 above)
Comment (stephenhand) [reply hrm-service/src/config/db.ts:12]: We did before, but now the support script is JS and this is TS. Also it's nice if the support scripts aren't dipping into application code if they can be kept separate. I thought overall a little duplication was worth it to have the separation

**Pull request merged**